cmd/common/image_functions.go

```diff
@@ -247,6 +248,13 @@ func UpdateForManifestWithoutSubjectToDelete(ctx context.Context, manifest acr.M
 		var manifestBytes []byte
 		manifestBytes, err := acrClient.GetManifest(ctx, repoName, *manifest.Digest)
 		if err != nil {
+			// Check if the error is autorest.DetailedError with status code not found
+			if detailedErr, ok := err.(autorest.DetailedError); ok && detailedErr.StatusCode == http.StatusNotFound {
+				fmt.Println("Manifest", *manifest.Digest, "not found, adding to delete candidates")
+				candidatesToDelete = append(candidatesToDelete, manifest)
```

Comment:
Why do we want to add this manifest to candidatesToDelete if it is not present? Instead, we should avoid the delete request completely

Reply:
Yes, the same manifests that return 404 may appeared in multiple purge cycles. 
But add to deleteCandidate seems not work as well, I just tried delete those manifests, it still appreared in next purge. 
So it seems that we can only wait all the registries to consistent state.
Updated the PR